### PR TITLE
update blog style and pinnedat setting

### PR DESCRIPTION
**Pull request by brain00021**
1. 修改語系黨
2. 增加pinned置頂功能
3. 限制置頂三篇
4. 新增table功能

### CODE REVIEW

File: src/components/blog/BlogPostTable.tsx [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+import pageMessages from '../../pages/translation'
+
+const StyledDiv = styled.div`
+  .ant-table-content {
+    padding: 0.25rem 1.5rem 2.5rem;
+  }
+`
+const StyledCover = styled.div<{ src: string }>`
+  position: relative;
+  // padding-top: ${900 / 16}%;
```
Comment (Ponchimeow) [inline src/components/blog/BlogPostTable.tsx:24]: NITS: remember remove unused comment

Comment (brain00021) [reply src/components/blog/BlogPostTable.tsx]: fixed it

File: src/components/blog/BlogPostTable.tsx [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+    const [searchAuthor, setSearchAuthor] = useState<string>('')
+    const [detailLoading, setDetailLoading] = useState(false)
+    const [blogDisplayData, setBlogDisplayData] = useState<BlogPostListColumn[]>([])
+    const [updatePostPinat] = useMutation<hasura.UPDATE_POST_PINNED_AT, hasura.UPDATE_POST_PINNED_ATVariables>(
+      UPDATE_POST_PINNED_AT,
+    )
+
+    const blogPostPinatNumber = blogPostData.filter(post => !!post.pinned_at).length
+
+    const handleUpload = (id: String | null, updateTime: Date | null, status: String) => {
```
Comment (Ponchimeow) [inline src/components/blog/BlogPostTable.tsx:86]: NITS: can decide upload status by updateTime, status props is unnecessary

Comment (brain00021) [reply src/components/blog/BlogPostTable.tsx:86]: not really sure what do you mean? do you mean i can check by another props?
i mean  blogPostPinatNumber will be set 0  when components reload if i set count  inside the "handleUpload"

File: src/components/blog/BlogPostTable.tsx [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+            <MoreOutlined style={{ fontSize: '20px' }} onClick={e => e.stopPropagation()} />
+          </Dropdown>
+        ),
+      },
+    ]
+    let displayColumns = postTableType === 'draft' ? columns.filter(items => items.key !== 'button') : columns
+
+    // if (error) {
+    //   return <div>{formatMessage(pageMessages['*'].fetchDataError)}</div>
+    // }
```
Comment (Ponchimeow) [inline src/components/blog/BlogPostTable.tsx:244]: NITS: remove unused comment~

File: src/components/blog/BlogPostTable.tsx [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+      )
+    } else {
+      return <div>{formatMessage(pageMessages['*'].fetchDataError)}</div>
+    }
+
+    // return searchTitle !== null ? (
+
+    // ) : (
+    //   <NoQuestionGroupBlock>{formatMessage(questionLibraryMessage.label.noQuestionGroup)}</NoQuestionGroupBlock>
+    // )
```
Comment (Ponchimeow) [inline src/components/blog/BlogPostTable.tsx:264]: NITS: remove unused comment~

File: src/hooks/blog.ts [hunk 5]
```diff
@@ -168,12 +174,16 @@ export const usePostCollection = () => {
           coverUrl: post.cover_url,
           videoUrl: post.video_url,
           views: post.views,
+          pinned_at: post.pinned_at,
```
Comment (sunnylochun) [inline src/hooks/blog.ts:177]: camelCase

File: src/components/blog/BlogPostTable.tsx [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+      let result
+      if (searchTitle === '' && searchAuthor === '') {
+        result = blogPostData
+      } else {
+        result = blogPostData.filter(
+          post => post.title?.includes(searchTitle) && post.authorName?.includes(searchAuthor),
+        )
+      }
+      setBlogDisplayData(result)
+    }, [searchTitle, searchAuthor, blogPostData])
```
Comment (sunnylochun) [inline src/components/blog/BlogPostTable.tsx:116]: NITS: try useMemo, more readable `to me`

Is this useEffect really needed ?

`example`
```
const filteredBlogPost = searchTitle === '' && searchAuthor === '' ? blogPostData : blogPostData.filter(
          post => post.title?.includes(searchTitle) && post.authorName?.includes(searchAuthor)
        )
...
<Table>
           ...
            dataSource={filteredBlogPost}
          />
        
```
is this work too?

File: src/components/blog/BlogPostTable.tsx (hunk 2 above)
Comment (Ponchimeow) [reply src/components/blog/BlogPostTable.tsx:86]: status 'update' is only used in the blogPostPinatNumber is 3 and the action is pinned a post
staus 'remove' never used
you can use updateTime is null or not to decide the 'update' or 'remove' action

Comment (brain00021) [reply src/components/blog/BlogPostTable.tsx]: fixed it

File: src/components/blog/BlogPostTable.tsx (hunk 3 above)
Comment (brain00021) [reply src/components/blog/BlogPostTable.tsx]: fixed it

File: src/components/blog/BlogPostTable.tsx (hunk 4 above)
Comment (brain00021) [reply src/components/blog/BlogPostTable.tsx]: fixed it

File: src/hooks/blog.ts (hunk 5 above)
Comment (brain00021) [reply src/hooks/blog.ts]: fixed it

File: src/components/blog/BlogPostTable.tsx (hunk 6 above)
Comment (brain00021) [reply src/components/blog/BlogPostTable.tsx]: fixed it

File: src/components/blog/BlogPostTable.tsx [hunk 7]
```diff
@@ -0,0 +1,257 @@
...
+
+    const blogPostPinatNumber = blogPostData.filter(post => !!post.pinnedAt).length
+
+    const handleUpload = (id: String | null, updateTime: Date | null = null, status: String | null = null) => {
+      if (blogPostPinatNumber === 3 && status) {
+        return message.warning(formatMessage(blogMessages.text.uploadPinnedAtLimmited))
+      }
+      setDetailLoading(true)
+
+      updatePostPinat({
```
Comment (Ponchimeow) [inline src/components/blog/BlogPostTable.tsx:92]: updatePostPinned?

File: src/components/blog/BlogPostTable.tsx [hunk 8]
```diff
@@ -0,0 +1,257 @@
...
+    const [updatePostPinat] = useMutation<hasura.UPDATE_POST_PINNED_AT, hasura.UPDATE_POST_PINNED_ATVariables>(
+      UPDATE_POST_PINNED_AT,
+    )
+
+    const blogPostPinatNumber = blogPostData.filter(post => !!post.pinnedAt).length
+
+    const handleUpload = (id: String | null, updateTime: Date | null = null, status: String | null = null) => {
+      if (blogPostPinatNumber === 3 && status) {
+        return message.warning(formatMessage(blogMessages.text.uploadPinnedAtLimmited))
+      }
```
Comment (Ponchimeow) [inline src/components/blog/BlogPostTable.tsx:89]: status is unnecessary
could be as follows
```js
    const handleUpload = (id: String | null, updateTime: Date | null = null) => {
      if (blogPostPinatNumber === 3 && updateTime) {
        return message.warning(formatMessage(blogMessages.text.uploadPinnedAtLimmited))
      }
      ...
      }
```

**Pull request merged**